core/utils/xml.ts

```diff
@@ -137,5 +146,11 @@ export function textToDomDocument(text: string): Document {
  */
 export function domToText(dom: Node): string {
   const oSerializer = new XMLSerializer();
-  return oSerializer.serializeToString(dom);
+  return sanitizeText(oSerializer.serializeToString(dom));
+}
+
+function sanitizeText(text: string) {
+  return text.replace(
+      INVALID_CONTROL_CHARS,
+      (match) => `&#x${match.charCodeAt(0).toString(16)};`);
```

Comment:
I'd recommend using decimal codes rather than hex.  Shorter (at least for smaller code points), faster, simpler.
`&#${match.charCodeAt(0)};`